x/compliance/keeper/keeper.go

```diff
@@ -310,34 +317,44 @@ func (k Keeper) GetVerificationDetailsByIssuer(ctx sdk.Context, userAddress sdk.
 
 // HasVerificationOfType checks if user has verifications of specific type (for example, passed KYC) from provided issuers.
 // If there is no provided expected issuers, this function will check if user has any verification of appropriate type.
-func (k Keeper) HasVerificationOfType(ctx sdk.Context, userAddress sdk.Address, expectedType types.VerificationType, expectedIssuers []sdk.Address) (bool, error) {
+func (k Keeper) HasVerificationOfType(ctx sdk.Context, userAddress sdk.Address, expectedType types.VerificationType, expirationTimestamp uint32, expectedIssuers []sdk.Address) (bool, error) {
 	// Obtain user address details
 	userAddressDetails, err := k.GetAddressDetails(ctx, userAddress)
 	if err != nil {
 		return false, err
 	}
 
-	// Filter verifications with expected type
-	var appropriateTypeVerifications []*types.Verification
-	for _, verification := range userAddressDetails.Verifications {
-		if verification.Type == expectedType {
-			appropriateTypeVerifications = append(appropriateTypeVerifications, verification)
-		}
+	// Use now as default expiration time if not given
+	if expirationTimestamp < 1 {
+		expirationTimestamp = uint32(time.Now().Unix())
```

Comment:
This conflicts with the current logic expirationTimestamp == 0 means indefinite

Reply:
Fixed it